addons/crm_livechat/models/chatbot_script_step.py

```diff
@@ -62,4 +70,39 @@ def _process_step_create_lead(self, discuss_channel):
         create_values.update(self._chatbot_crm_prepare_lead_values(
             discuss_channel, customer_values['description']))
 
-        self.env['crm.lead'].create(create_values)
+        return self.env['crm.lead'].create(create_values)
+
+    def _process_step_create_lead_and_forward(self, discuss_channel, posted_message):
+        lead = self._process_step_create_lead(discuss_channel)
+        team = self.crm_team_id
+        domain = literal_eval(team.assignment_domain or "[]")
+        lead = lead.filtered_domain(domain)
```

Comment:
If the team is determined by the script step, do we really need to check the assignment_domain?
I'd assume all leads are considered valid in this case as the purpose here is not to find the proper team, but to find an available member.

When `lead.team_id` is empty (none set on the step), only then we should try to find the appropriate team (with hopefully using standard crm methods, but I'm not too familiar with them).

Reply:
Don't use assignment domain and quota, this is used for automatic lead assign. If you want to use heavy assign, you should remove all that and let the cron do its job. It is much more tailored for that.

Reply:
Indeed the `assignment_domain` is useful if the `auto_assign_cron` is enabled. And using `assignment_domain` doesn't help aswell.

Relying on the `cron` might not be useful here either. Therefore, I think it makes sense to assign the partner directly in this context. That said, we should consider using the `assignment_quota`` to ensure that users are not assigned beyond their maximum capacity.

Reply:
Yep if we really want to force assign, it is going to be used. Once everything is ready, please ping so that we check a bit from crm point of view :) .